crane/models.py

```diff
@@ -100,18 +98,30 @@ def is_disconnected(self):
             return True
 
     def check_preconditions(self):
-        from . import settings  # avoiding circular imports
-
+        from . import hooks, settings  # avoiding circular imports
         try:
+            self.repo = git.Repo(environ['CI_PROJECT_DIR'])
             self.new_commit
```

Comment:
These two should not be in the same `try` block, as far as I can tell the `Repo` call can raise only `git.NoSuchPathError` while accessing `self.new_commit` can raise only `gitdb.exc.BadName`.